src/Platforms/AbstractPlatform.php

```diff
@@ -794,6 +796,16 @@ private function buildCreateTableSQL(Table $table, bool $createForeignKeys): arr
         $options['primary']           = [];
 
         foreach ($table->getIndexes() as $index) {
+            if ($index->isFunctional() && ! $this->supportsFunctionalIndex()) {
+                throw new InvalidArgumentException(sprintf(
+                    'Index "%s" on table "%s" contains a functional part, ' .
+                    'but platform "%s" does not support functional indexes.',
+                    $index->getName(),
+                    $table->getName(),
+                    substr(static::class, (int) strrpos(static::class, '\\') + 1),
```

Comment:
The platform could be overridden in userland. Let's print the FQCN.

```suggestion
                    get_debug_type($this),
```

Reply:
done